Trakt/ScheduledTasks/SyncFromTraktTask.cs

```diff
@@ -336,42 +336,41 @@ private async Task SyncTraktDataForUser(Jellyfin.Data.Entities.User user, double
-                        if (matchedWatchedEpisode == null && matchedWatchedEpisodeHistory != null)
+                        // Prepend a check if the matched episode is on a rewatch cycle and
+                        // discard it if the last play date was before the reset date
+                        if (lastWatchedEpisodeHistory != null
+                            && tLastReset != null
+                            && DateTime.TryParse(lastWatchedEpisodeHistory.WatchedAt, out var lastPlayedValue)
+                            && lastPlayedValue < tLastReset)
                         {
-                            // Find watched season and episode via history match
-                            _logger.LogDebug("Using history to match season and episode for user {User} for {Data}", user.Username, GetVerboseEpisodeData(episode));
-                            matchedWatchedEpisode = matchedWatchedSeason.Episodes.FirstOrDefault(tEpisode => tEpisode.Number == matchedWatchedEpisodeHistory.Episode.Number);
+                            lastWatchedEpisodeHistory = null;
                         }
 
-                        // Fallback procedure to find match by using episode history, without checking the season (episode can belong to different season)
-                        if (matchedWatchedEpisode == null && matchedWatchedEpisodeHistory != null)
+                        if (lastWatchedEpisodeHistory != null)
                         {
-                            // Find watched episode via history match
-                            _logger.LogDebug("Using history to match episode for user {User} for {Data}", user.Username, GetVerboseEpisodeData(episode));
-                            foreach (var season in matchedWatchedShow.Seasons)
+                            _logger.LogDebug("Episode is in watched list of user {User}: {Data}", user.Username, GetVerboseEpisodeData(episode));
+
+                            episodeWatched = true;
+                            DateTime? tLastPlayed = null;
+                            if (DateTime.TryParse(lastWatchedEpisodeHistory.WatchedAt, out var lastWatchedValue))
                             {
-                                matchedWatchedEpisode = season.Episodes.FirstOrDefault(tEpisode => tEpisode.Number == matchedWatchedEpisodeHistory.Episode.Number);
-                                if (matchedWatchedEpisode != null)
-                                {
-                                    break; // stop when found in a season
-                                }
+                                tLastPlayed = lastWatchedValue;
                             }
+
+                            // Update episode data
+                            changed = UpdateEpisodeData(user, episode, userData, tLastPlayed, matchedWatchedEpisodeHistory.Count());
                         }
+                    }
+                    else if (matchedWatchedSeason != null)
```

Comment:
```suggestion
                    else if (matchedWatchedSeason is not null)
```

Reply:
Same remark here: `is not null` is not used at all in current file.

Reply:
We've changed the server code to `is null` recently but it would make more sense to change the whole plugin code at once in a separate PR, so feel free to ignore those comments.

Reply:
Ok, then I think I resolved all your requested changes.